porch/pkg/registry/porch/packagerevisionresources.go

```diff
@@ -170,3 +172,89 @@ func (r *packageRevisionResources) Update(ctx context.Context, name string, objI
 	}
 	return created, false, nil
 }
+
+// Create implements the Creater interface.
+func (r *packageRevisionResources) Create(ctx context.Context, runtimeObject runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	ctx, span := tracer.Start(ctx, "packageRevisionResources::Create", trace.WithAttributes())
+	defer span.End()
+
+	ns, namespaced := genericapirequest.NamespaceFrom(ctx)
+	if !namespaced {
+		return nil, apierrors.NewBadRequest("namespace must be specified")
+	}
+
+	obj, ok := runtimeObject.(*api.PackageRevisionResources)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected PackageRevisionResources object, got %T", runtimeObject))
+	}
+
+	// TODO: Accpept some form of client-provided name, for example using GenerateName
+	// and figure out where we can store it (in Kptfile?). Porch can then append unique
+	// suffix to the names while respecting client-provided value as well.
+	if obj.Name != "" {
+		klog.Warningf("Client provided metadata.name %q", obj.Name)
+	}
+
+	repositoryName := obj.Spec.RepositoryName
+	if repositoryName == "" {
+		return nil, apierrors.NewBadRequest("spec.repositoryName is required")
+	}
+
+	var repositoryObj configapi.Repository
+	repositoryID := types.NamespacedName{Namespace: ns, Name: repositoryName}
+	if err := r.coreClient.Get(ctx, repositoryID, &repositoryObj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, apierrors.NewNotFound(configapi.KindRepository.GroupResource(), repositoryID.Name)
+		}
+		return nil, apierrors.NewInternalError(fmt.Errorf("error getting repository %v: %w", repositoryID, err))
+	}
+
+	fieldErrors := r.createStrategy.Validate(ctx, runtimeObject)
+	if len(fieldErrors) > 0 {
+		return nil, apierrors.NewInvalid(api.SchemeGroupVersion.WithKind("PackageRevision").GroupKind(), obj.Name, fieldErrors)
+	}
+
+	// We currently create the PackageRevision, and then attach the resources
+	// TODO: Can we streamline this - create the package with its resources, or maybe return the draft?
```

Comment:
Could we address this by creating a new task type that could add the resources to the package as apart of creating the initial revision?

Reply:
We could, but I worry that it would make the model more complex, in that now some resources are directly created in PackageRevision.  That said, we know that patches share some elements of this problem in the reclone-and-replay approach.